src/runner/fullJSRunner.ts

```diff
@@ -0,0 +1,106 @@
+/* eslint-disable @typescript-eslint/no-unused-vars */
+import { Options, parse } from 'acorn'
+import { generate } from 'astring'
+import * as es from 'estree'
+
+import { IOptions, Result } from '..'
+import { NATIVE_STORAGE_ID } from '../constants'
+import { FatalSyntaxError } from '../parser/parser'
+import { evallerReplacer, getBuiltins, prefixModule, transpile } from '../transpiler/transpiler'
+import { Context } from '../types'
+import * as create from '../utils/astCreator'
+import { NativeStorage } from './../types'
+import { toSourceError } from './errors'
+import { appendModulesToContext, resolvedErrorPromise } from './utils'
+
+const FULL_JS_PARSER_OPTIONS: Options = {
+  sourceType: 'module',
+  ecmaVersion: 'latest',
+  locations: true
+}
+
+/**
+ * Parse code string into AST
+ * - any errors in the process of parsing will be added to the context
+ *
+ * @param code
+ * @param context
+ * @returns AST of code if there are no syntax errors, otherwise undefined
+ */
+function parseFullJS(code: string, context: Context): es.Program | undefined {
+  let program: es.Program | undefined
+  try {
+    program = parse(code, FULL_JS_PARSER_OPTIONS) as unknown as es.Program
+  } catch (error) {
+    if (error instanceof SyntaxError) {
+      const loc = (error as any).loc
+      const location = {
+        start: { line: loc.line, column: loc.column },
+        end: { line: loc.line, column: loc.column + 1 }
+      }
+      context.errors.push(new FatalSyntaxError(location, error.toString()))
+    }
+  }
+
+  return program
+}
+
+function fullJSEval(code: string, nativeStorage: NativeStorage, moduleParams: any): any {
+  if (nativeStorage.evaller) {
+    return nativeStorage.evaller(code)
+  } else {
+    return eval(code)
+  }
+}
+
+function preparePrelude(context: Context): es.Statement[] {
+  if (context.prelude === null) {
+    return []
+  }
+  const prelude = context.prelude
+  context.prelude = null
+  const program: es.Program = parseFullJS(prelude, context)!
+
+  return program.body as es.Statement[]
+}
+
+export async function fullJSRunner(
+  code: string,
+  context: Context,
+  options: Partial<IOptions> = {}
+): Promise<Result> {
+  // parse + check for syntax errors
+  const program: es.Program | undefined = parseFullJS(code, context)
+  if (!program) {
+    return resolvedErrorPromise
+  }
+
+  // prelude & builtins
+  // TODO resolve repeated declaration in prelude and builtins
+  context.nativeStorage.builtins.delete('list_to_stream')
```

Comment:
there seems to be a `list_to_stream` declaration in both `builtins` (during context creation) and stream prelude. I'm not sure if it's okay to remove the declaration in the prelude 🤔

Reply:
Removed in [7ce95b6](https://github.com/source-academy/js-slang/pull/1206/commits/7ce95b647c6e613704f3e8fbea0439fb8af1ee87)